modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/processors/SW360EnricherImpl.java

```diff
@@ -73,12 +76,15 @@ private void mapExternalIdsOnSW360Release(SW360Release sw360Release, Artifact ar
         mapCoordinates(sw360Release)
                 .forEach(artifact::addFact);
 
-        artifact.addFact(new DeclaredLicenseInformation(makeLicenseStatementFromString(sw360Release.getDeclaredLicense())));
```

Comment:
This can be solved in a simpler way:
```suggestion
        Optional.ofNullable(sw360Release.getDeclaredLicense())
            .map(this::makeLicenseStatementFromString)
            .map(DeclaredLicenseInformation::new)
            .ifPresent(artifact::add);
        Optional.ofNullable(sw360Release.getObservedLicense())
            .map(this::makeLicenseStatementFromString)
            .map(ObservedLicenseInformation::new)
            .ifPresent(artifact::add);
```

This results in less code and does not use reflection.

Reply:
This is so ugly :-)